File: core/src/com/jupiter/europa/entity/ability/TargetInfo.java
package com.jupiter.europa.entity.ability;

/**
 * Created by nathan on 5/21/15.
 */
public class TargetInfo<T extends Target> {

    // Enumerations
    public enum TargetType {
        TILE,
        ENTITY;
    }


    // Fields
    private final TargetType type;
    private final TargetFilter<T> filter;


    // Initialization
    public TargetInfo(TargetType type, TargetFilter<T> filter) {
        this.type = type;
        this.filter = filter;
    }


    @FunctionalInterface
    public interface TargetFilter<T extends Target> {
        boolean accept(Target target);
    }
}

File: core/src/com/jupiter/europa/entity/ability/TargetInfo.kt
package com.jupiter.europa.entity.ability

/**
 * Created by nathan on 5/21/15.
 */
public class TargetInfo<T : Target>// Initialization
(private val type: TargetInfo.TargetType, private val filter: TargetInfo.TargetFilter<T>) {

    // Enumerations
    public enum class TargetType {
        TILE,
        ENTITY
    }


    FunctionalInterface
    public interface TargetFilter<T : Target> {
        public fun accept(target: Target): Boolean
    }
}
